yew-functional/src/use_context_hook.rs

```diff
@@ -58,45 +55,35 @@ impl<T: Clone + PartialEq> ContextProvider<T> {
     }
 }
 
-impl<T: Clone + PartialEq + 'static> Component for ContextProvider<T> {
-    type Message = ();
+impl<T: PartialEq + 'static> Component for ContextProvider<T> {
+    type Message = Weak<ConsumerCallback<T>>;
     type Properties = ContextProviderProps<T>;
 
-    fn create(props: Self::Properties, _link: ComponentLink<Self>) -> Self {
+    fn create(_ctx: &Context<Self>) -> Self {
         Self {
-            children: props.children,
-            context: Rc::new(props.context),
             consumers: RefCell::new(Vec::new()),
         }
     }
 
-    fn update(&mut self, _msg: Self::Message) -> bool {
-        true
+    fn update(&mut self, _ctx: &Context<Self>, msg: Self::Message) -> bool {
+        self.subscribe_consumer(msg);
+        false
     }
 
-    fn change(&mut self, props: Self::Properties) -> bool {
-        let should_render = if self.children == props.children {
-            false
-        } else {
-            self.children = props.children;
-            true
-        };
-
-        let new_context = Rc::new(props.context);
-        if self.context != new_context {
-            self.context = new_context;
-            self.notify_consumers();
+    fn changed(&mut self, ctx: &Context<Self>, new_props: &Self::Properties) -> bool {
+        if ctx.props.context != new_props.context {
+            self.notify_consumers(new_props.context.clone());
```

Comment:
I just have to follow Rust recommendations :P
```suggestion
            self.notify_consumers(Rc::clone(&new_props.context));
```